File: library/src/main/java/com/mikepenz/materialdrawer/model/interfaces/IProfile.java
package com.mikepenz.materialdrawer.model.interfaces;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.support.annotation.DrawableRes;

import com.mikepenz.fastadapter.IIdentifyable;
import com.mikepenz.iconics.typeface.IIcon;
import com.mikepenz.materialdrawer.holder.ImageHolder;
import com.mikepenz.materialdrawer.holder.StringHolder;

/**
 * Created by mikepenz on 03.02.15.
 */
public interface IProfile<T> extends IIdentifyable<T> {
    T withName(String name);

    StringHolder getName();

    T withEmail(String email);

    StringHolder getEmail();

    T withIcon(Drawable icon);

    T withIcon(Bitmap bitmap);

    T withIcon(@DrawableRes int iconRes);

    T withIcon(String url);

    T withIcon(Uri uri);

    T withIcon(IIcon icon);

    ImageHolder getIcon();

    T withSelectable(boolean selectable);

    boolean isSelectable();
}

File: library/src/main/java/com/mikepenz/materialdrawer/model/interfaces/IProfile.kt
package com.mikepenz.materialdrawer.model.interfaces

import androidx.annotation.StringRes
import com.mikepenz.fastadapter.IIdentifyable
import com.mikepenz.materialdrawer.holder.StringHolder

/**
 * Defines a general [IProfile] to be displayed in the [com.mikepenz.materialdrawer.widget.MaterialDrawerSliderView] with the [com.mikepenz.materialdrawer.widget.AccountHeaderView]
 */
interface IProfile : IIdentifyable, Nameable, Iconable, Selectable, Tagable, Describable

@Deprecated("Please consider to replace with the actual property setter")
fun <T : IProfile> T.withEmail(@StringRes emailRes: Int): T {
    this.description = StringHolder(emailRes)
    return this
}

@Deprecated("Please consider to replace with the actual property setter")
fun <T : IProfile> T.withEmail(email: String?): T {
    this.description = StringHolder(email)
    return this
}
